### DISCUSSION TITLE
PHP Fatal error:  Uncaught Error: Class 'Phalcon\Di\FactoryDefault\Cli' not found

### DISCUSSION THREAD

**Discussion by FabrizioCorporaTO** [phalcon/phalcon#342] [General]
hi, I'm inserting the phalcon cli in a dev environment which has a version of php/phalcon:
PHP Version 7.0.32-0ubuntu0.16.04.1
Author Phalcon Team and contributors
Version 3.1.2
Build Date Apr 13 2017 11:46:10
Powered by Zephir Version 0.9.7-3cb5f8f376

unfortunately running returns this error.
PHP Fatal error:  Uncaught Error: Class 'Phalcon\Di\FactoryDefault\Cli' not found in /store/www/.../app/cli.php:20 

Instead on a local environment it works correctly:

PHP Version 7.0.30-1+ubuntu16.04.1+deb.sury.org+1
Author Phalcon Team and contributors
Version 3.4.0
Build Date Jun 6 2018 04:43:19
Powered by Zephir Version 0.10.10-d1b4cc68d9

I did several searches and it seems to exist in an older version than mine:
https://devdocs.io/phalcon~3/api/phalcon_di_factorydefault_cli

https://phalcon-php-framework-documentation.readthedocs.io/en/latest/api/Phalcon_Di_FactoryDefault_Cli.html

https://docs.w3cub.com/phalcon~2/api/phalcon_di_factorydefault_cli

I honestly don't know whether to trust these documentations.
I await your news.
Thank you.

**Comment by borisdelev** [comment]
Amm, not sure, but they are differences between Phalcon 3.1 and 3.4. Maybe this is the issue. Can you make it both to 3.4?

PS: this discussion must be placed in cphalcon, but anyway

**Reply by FabrizioCorporaTO** [reply]
hi, thanks for the reply.

I apologize if I misplaced this issue.


I haven't tried it yet in the dev environment with 3.4 because we are waiting for the update.


i will try as soon as possible.

**Reply by borisdelev** [reply]
Hope this is the issue. Anyway try Phalcon v5. Its great 😊

**Reply by niden** [reply]
v3 is many years behind the current version. I would strongly suggest you upgrade your code if possible.

v3 did not change anything in terms of interfaces between each sub-version (3.1,3.2 etc.).

When the class is not "there" there is a problem with your installation. Try putting `phpinfo()` in one of your web pages and check if phalcon is present. Also as CLI you can create a small script that will "dump" the phalcon version on screen. Run it and see if that works.

It is often that the web server and the CLI have different  php.ini files and you need to enable phalcon on both.

**Reply by FabrizioCorporaTO** [reply]
Hello, Thanks for the reply.


These are the info I found with phpinfo related to phalcon:

ENV LOCAL:
phalcon
Web framework delivered as a C-extension for PHP
phalcon	enabled
Author	Phalcon Team and contributors
Version	3.4.0
Build Date	Jun 6 2018 04:43:19
Powered by Zephir	Version 0.10.10-d1b4cc68d9
Directive	Local Value	Master Value
phalcon.db.escape_identifiers	On	On
phalcon.db.force_casting	Off	Off
phalcon.orm.cast_on_hydrate	Off	Off
phalcon.orm.column_renaming	On	On
phalcon.orm.disable_assign_setters	Off	Off
phalcon.orm.enable_implicit_joins	On	On
phalcon.orm.enable_literals	On	On
phalcon.orm.events	On	On
phalcon.orm.exception_on_failed_save	Off	Off
phalcon.orm.ignore_unknown_columns	Off	Off
phalcon.orm.late_state_binding	Off	Off
phalcon.orm.not_null_validations	On	On
phalcon.orm.update_snapshot_on_save	On	On
phalcon.orm.virtual_foreign_keys	On	On

/etc/php/7.0/fpm/conf.d/20-phalcon.ini,


ENV DEV:
phalcon
Web framework delivered as a C-extension for PHP
phalcon	enabled
Author	Phalcon Team and contributors
Version	3.1.2
Build Date	Apr 13 2017 11:46:10
Powered by Zephir	Version 0.9.7-3cb5f8f376
Directive	Local Value	Master Value
phalcon.db.escape_identifiers	On	On
phalcon.db.force_casting	Off	Off
phalcon.orm.cast_on_hydrate	Off	Off
phalcon.orm.column_renaming	On	On
phalcon.orm.enable_implicit_joins	On	On
phalcon.orm.enable_literals	On	On
phalcon.orm.events	On	On
phalcon.orm.exception_on_failed_save	Off	Off
phalcon.orm.ignore_unknown_columns	Off	Off
phalcon.orm.late_state_binding	Off	Off
phalcon.orm.not_null_validations	On	On
phalcon.orm.virtual_foreign_keys	On	On

/etc/php/7.0/apache2/conf.d/30-phalcon.ini

the only difference I notice is the priority which on DEV is 30.

**Comment by FabrizioCorporaTO** [comment]
i fixed the problem after upgrading to phalcon version 3.4.
